pkg/controller/actuator_reconcile.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extension-runtime-gvisor/pkg/charts"
+
+	resourcemanagerv1alpha1 "github.com/gardener/gardener-resource-manager/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener-resource-manager/pkg/manager"
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	GVisorInstallationSecretName          = "extension-runtime-gvisor-installation"
+	GVisorSecretName                      = "extension-runtime-gvisor"
+	GVisorInstallationManagedResourceName = "extension-runtime-gvisor-installation"
+	GVisorManagedResourceName             = "extension-runtime-gvisor"
```

Comment:
These names don't include the worker pool name. What if multiple pools specify the gvisor container runtime? Yes, it might work, but it is a little odd that multiple independent reconciliations all work on the same object.

Reply:
[Please see above](https://github.com/gardener/gardener-extension-runtime-gvisor/pull/3#discussion_r405450156).
The GVisorManagedResourceName and GVisorSecretName are worker pool independent resources. When there is a gVisor installation required, it will be installed as a prerequisite. When no worker-pool requires gVisor any more, it will be removed at last. 

The installation managed resource and secret are worker-pool dependent and [have a name that includes the worker-pool name](https://github.com/gardener/gardener-extension-runtime-gvisor/blob/597dc592335a27b037d068d04dd95ad1a0415258/pkg/controller/actuator_reconcile.go#L88).

Reply:
Yes, I see. I also see that duplicating the resources is ugly as well. I don't have a good proposal now. If you think there is no better solution we can also keep it like this for now...

Reply:
Duplicating the resources is also no option as deinstallation  temporarely removes the RuntimeClass and thus breaks scheduling gVisor pods - plus duplicate Service, Accounts psp, rbac.
Yes I would go like this for. When testing it I have not seen a drawback.
Of course I am open to changing it in the future if we discover a major drawback.

Reply:
Fine...